test/e2e_env/kubernetes/gateway/delegated/meshretry.go

```diff
@@ -0,0 +1,86 @@
+package delegated
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshretry/api/v1alpha1"
+	"github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func MeshRetry(config *Config) func() {
+	GinkgoHelper()
+
+	return func() {
+		meshRetryPolicy := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshRetry
+metadata:
+  name: mr
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          numRetries: 6
+          retryOn: ["503"]
+`, config.CpNamespace, config.Mesh)
+
+		framework.E2EAfterEach(func() {
+			Expect(framework.DeleteMeshResources(
+				kubernetes.Cluster,
+				config.Mesh,
+				v1alpha1.MeshRetryResourceTypeDescriptor,
+			)).To(Succeed())
+		})
+
+		It("should retry on HTTP connection failure", func() {
```

Comment:
connection failure would be `ConnectFailure` right? 

```suggestion
		It("should retry on HTTP 503", func() {
```